modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -1113,6 +1113,18 @@ private void releaseWalKeys(long segmentIdx) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public void onDoneAfterTopologyUnlock(GridDhtPartitionsExchangeFuture fut) {
+        if (fut.activateCluster() || fut.localJoinExchange()) {
```

Comment:
What if the current node joins an inactive cluster?

Reply:
If a node joins an inactive cluster, no PME occurs.